conans/client/tools/files.py

```diff
@@ -374,3 +375,45 @@ def remove_files_by_mask(directory, pattern):
                 os.unlink(fullname)
                 removed_names.append(os.path.relpath(fullname, directory))
     return removed_names
+
+
+def fix_symlinks(conanfile, raise_if_error=False):
+    """ Fix the symlinks in the conanfile.package_folder: make symlinks relative and remove
+        those links to files outside the package (it will print an error, or raise
+        if 'raise_if_error' evaluates to true).
+    """
+    offending_files = []
+
+    def work_on_element(dirpath, element, token):
+        fullpath = os.path.join(dirpath, element)
+        if os.path.islink(fullpath):
+            link_target = os.readlink(fullpath)
+            link_abs_target = os.path.join(dirpath, link_target)
+            link_rel_target = os.path.relpath(link_abs_target, conanfile.package_folder)
+            if link_rel_target.startswith('..'):
+                offending_file = os.path.relpath(fullpath, conanfile.package_folder)
+                offending_files.append(offending_file)
+                conanfile.output.error("{token} '{item}' links to a {token} outside the package, "
+                                       "it's been removed.".format(item=offending_file, token=token))
+                os.unlink(fullpath)
+            elif not os.path.exists(link_abs_target):
+                # This is a broken symlink. Failure is controlled by config variable
+                #  'general.skip_broken_symlinks_check'. Do not fail here.
+                offending_file = os.path.relpath(fullpath, conanfile.package_folder)
+                offending_files.append(offending_file)
+                conanfile.output.error("{token} '{item}' links to a path that doesn't exist, it's"
+                                       " been removed.".format(item=offending_file, token=token))
+                os.unlink(fullpath)
+            elif link_target != link_rel_target:
+                os.unlink(fullpath)
```

Comment:
what about adding a info message, or even a log info? After running package, Conan reports how many files were copied, but we don't know how many are symbolic links. This method could empathize the symlink update.

Reply:
IMO that output is not needed: you are writing the recipe, you are creating the package, you know you are calling this tool and the message here will tell you only about updated symlinks (removed ones are already reported), IMO it would be more like a debug-level output (and we don't have that level in the output).

Reply:
if we want to add an output, it doesn't belong here. conan might be running with different output formatters (e.g. human-readable, json, HTML, etc.). this function probably should just reference helpers from `conan_command_output`. otherwise we may run into weird situation receiving human-readable messages running commands on part of CI scripts via `--json` (or `--html`)